File: src/main/java/com/giant/module/employee/entity/EmployeeSerial.java
package com.giant.module.employee.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "\"serial\"", schema = "employee")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class EmployeeSerial {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long serialId;

    @Column(
            name = "serial_name",
            nullable = false,
            unique = true
    )
    private String serialName;

    @Column(name = "serial_value", nullable = false)
    private Long serialValue;
}

File: src/main/kotlin/com/giant/module/employee/entity/EmployeeSerial.kt
package com.giant.module.employee.entity

import jakarta.persistence.*

@Entity
@Table(name = "\"serial\"", schema = "employee")
class EmployeeSerial(
    @Column(name = "serial_name", nullable = false, unique = true)
    var serialName: String,

    @Column(name = "serial_value", nullable = false)
    var serialValue: Long
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    val serialId: Long? = null

    fun increaseSerialValue(table: EmployeeSerial) {
        this.serialValue = table.serialValue + 1
    }

    companion object {
        fun create(serialName: String, serialValue: Long): EmployeeSerial {
            return EmployeeSerial(serialName, serialValue)
        }
    }
}